### PR TITLE
Create UVM = OFF build

### PR DESCRIPTION
**Pull request by ZUUL42**
@trilinos/framework 
CC: @kddevin 

## Motivation

This creates a UVM = OFF specific build using CUDA 10.1.105.

4 packages were turned off, panzer, stokhos, intrepid2, & trilinoscouplings in order to get it to build cleanly.
A significant number of tests had to be turned off in order to get this build to test cleanly. However, this will give us a place to start.

As package teams make modification to get tests to pass, they can be turned back on by removing the associated line in:
[`cmake\std\PullRequestLinuxCuda10.1.105uvmTestingSettings.cmake`](https://github.com/trilinos/Trilinos/blob/develop/cmake/std/PullRequestLinuxCuda10.1.105uvmTestingSettings.cmake)
Teams can do this, just request a review from any of the Framework team and the test will be reenabled for this build.

Associated Issue: #8397

## Testing

This has been tested manually with the new Cuda 10.1.105 uvm job. See #8397.
It shouldn't affect other builds.

The next step will be to add it to the dev->master build set.

### CODE REVIEW

**Comment by csiefer2** [comment]
Is there a reason to disable the failing tests?  Why not just leave them on and have them fail?

**Comment by jwillenbring** [comment]
> Is there a reason to disable the failing tests (since they number in the hundreds)? Why not just leave them on and have them fail?

We have been asked to set up a PR build that protects what is passing and add to this over time. There cannot be any failing tests in a PR build.

**Comment by csiefer2** [comment]
> We have been asked to set up a PR build that protects what is passing and add to this over time. There cannot be any failing tests in a PR build.

Ah, it didn't say up top that it was for a PR build!  That makes much more sense!

As a warning:  People are not going to remember to re-enable tests when they get fixed.  Periodic reminders to package leads would be advised.

**Comment by ZUUL42** [comment]
@csiefer2 I didn't note it explicitly. All PR builds as well as dev->master builds must build cleanly and pass all tests. Otherwise the PR build merge to develop and/or the merge to master won't happen. This will also prevent any backsliding of progress. As a dev->master build, Framework and anyone else watching the dev->master builds will be alerted when anything fails.
Then teams can turn on tests as they make modifications and I can periodically go in and run tests to see if additional tests are passing.
If we promote this to a PR build, it will prevent any changes that break a test to be submitted.

While I admit it is a _lot_ of tests disabled, it is a place to start and hold ground from here as packages advance Trilinos to be UVM free.

**Comment by csiefer2** [comment]
@ZUUL42 I suspect what you'll actually have is packages updating nothing until all of their tests are passing.  Less work for the developer, but still useful.

File: cmake/std/sems/PullRequestCuda10.1.105uvmTestingEnv.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+module load git/2.10.1
+module load devpack/20190404/openmpi/4.0.1/gcc/7.2.0/cuda/10.1.105
+module swap openblas/0.3.4/gcc/7.4.0 netlib/3.8.0/gcc/7.2.0
+#export OMPI_CXX=`which g++`
+current_dir=`dirname $BASH_SOURCE`
+Trilinos_dir=`realpath $current_dir/../../..`
+export OMPI_CXX=${Trilinos_dir}/packages/kokkos/bin/nvcc_wrapper
+export OMPI_CC=`which gcc`
+export OMPI_FC=`which gfortran`
+export CUDA_LAUNCH_BLOCKING=1
```
Comment (jwillenbring) [inline cmake/std/sems/PullRequestCuda10.1.105uvmTestingEnv.sh:15]: Why is launch blocking turned on?

File: cmake/std/trilinosprhelpers/unittests/trilinos_pr_test.ini [hunk 2]
```diff
@@ -306,6 +307,21 @@ unsetenv PATH_TMP:
...
 
 
+[Trilinos_pullrequest_cuda_10.1.105_uvm]
+module-load git: 2.10.1
+module-load devpack: 20190404/openmpi/4.0.1/gcc/7.2.0/cuda/10.1.105
+module-swap openblas/0.3.4/gcc/7.4.0: netlib/3.8.0/gcc/7.2.0
+setenv OMPI_CXX: ${WORKSPACE}/Trilinos/packages/kokkos/bin/nvcc_wrapper
+setenv OMPI_CC: ${CC}
+setenv OMPI_FC: ${FC}
+setenv CUDA_LAUNCH_BLOCKING: 1
```
Comment (jwillenbring) [inline cmake/std/trilinosprhelpers/unittests/trilinos_pr_test.ini:317]: Why is launch blocking on?

Comment (ZUUL42) [reply cmake/std/trilinosprhelpers/unittests/trilinos_pr_test.ini:317]: I started with a copy of the current CUDA 10.1.105 build. We can turn it off.

**Comment by csiefer2** [comment]
On second thought, using this as a PR test might not work with the Tpetra team's approach to UVM removal --- we're contemplating intentionally completely breaking downstream builds because we're going to change interfaces needed to support use w/o UVM.  This is not set in stone, but you should really talk about it with @kddevin.

**Comment by jwillenbring** [comment]
@ZUUL42 I think what happened here is like just the fact that there are now conflicts between this branch and the current develop. If you resolve those, testing may resume as normal.

**Review by jwillenbring — APPROVED** [review]
This looks good.

**Pull request merged**